src/components/OrderableList/OrderableList.tsx

```diff
@@ -0,0 +1,261 @@
+import { useDroppableCollection } from "@react-aria/dnd";
+import { useGrid } from "@react-aria/grid";
+import { ListKeyboardDelegate } from "@react-aria/selection";
+import { mergeProps } from "@react-aria/utils";
+import { Item } from "@react-stately/collections";
+import { useDraggableCollectionState, useDroppableCollectionState } from "@react-stately/dnd";
+import { GridCollection, useGridState } from "@react-stately/grid";
+import { useListState } from "@react-stately/list";
+import { DropTarget, ItemDropTarget } from "@react-types/shared";
+import React, { useRef } from "react";
+import { GridNode } from "@react-types/grid";
+import { InsertionIndicator } from "./InsertionIndicator";
+import { CollectionItem } from "./CollectionItem";
+import { ItemDragState, OrderableListContainerProps, OrderableListItem, OrderableListProps } from "./types";
+import { useMemoizedId } from "@hooks/useMemoizedId";
+
+export const OrderableList = <T extends object>({
+    onMove,
+    items,
+    renderContent,
+    dragDisabled,
+    disableTypeAhead,
+}: OrderableListProps<T>) => {
+    return (
+        <OrderableListContainer
+            onMove={onMove}
+            renderContent={renderContent}
+            items={items}
+            disableTypeAhead={disableTypeAhead}
+            dragDisabled={dragDisabled}
+        >
+            {(item: OrderableListItem<T>) => (
+                <Item textValue={item.alt}>
+                    <></>
+                </Item>
+            )}
+        </OrderableListContainer>
+    );
+};
+
+export const OrderableListContainer = <T extends object>(props: OrderableListContainerProps<T>) => {
+    const gridRef = useRef<HTMLDivElement>(null);
+
+    const state = useListState(props);
+    const keyboardDelegate = new ListKeyboardDelegate(state.collection, new Set(), gridRef);
+
+    const gridState = useGridState({
+        selectionMode: "single",
+        collection: new GridCollection<OrderableListItem<T>>({
+            columnCount: 1,
+            items: [...state.collection].map((item) => ({
+                ...item,
+                childNodes: [
+                    {
+                        key: `cell-${item.key}`,
+                        type: "cell",
+                        index: 0,
+                        value: item.value,
+                        level: 0,
+                        rendered: item.rendered,
+                        textValue: item.textValue,
+                        hasChildNodes: false,
+                        childNodes: [],
+                    },
+                ],
+            })),
+        }),
+    });
+
+    const { gridProps } = useGrid(
+        {
+            "aria-label": "Reorderable list",
+            focusMode: "cell",
+        },
+        gridState,
+        gridRef,
+    );
+
+    // Use a random drag type so the items can only be reordered within this list and not dragged elsewhere.
+    const dragType = useMemoizedId();
```

Comment:
Maybe change to a more suitable name like `dragTypeId`

Reply:
done